### PR TITLE
[magnum] Properly deploy plugins

### PR DESCRIPTION
**Pull request by Squareys**
Hi @ras0219-msft !

I finally fixed the plugin deployment of the magnum package (see https://github.com/mosra/magnum/issues/235)! As this is the first time I came in contact with Powershell, you may want to have a close look at that.

Also, to be able to build `--head` with upcoming features properly, I added the repective features and incremented the CONTROL-file version.

@mosra may want to have a quick glance at the depedencies of new features.

Thanks in advance,
Jonathan.

### CODE REVIEW

**Comment by Squareys** [comment]
@mosra Done with the changes discussed on gitter! :)

File: ports/magnum-plugins/portfile.cmake [hunk 1]
```diff
@@ -41,6 +43,7 @@ vcpkg_configure_cmake(
     OPTIONS
         ${_COMPONENT_FLAGS}
         -DBUILD_STATIC=${BUILD_STATIC}
+        -DBUILD_PLUGINS_STATIC=${DBUILD_PLUGINS_STATIC}
```
Comment (mosra) [inline ports/magnum-plugins/portfile.cmake]: Hmm, maybe just have a single `BUILD_PLUGINS_STATIC` variable and set it to both? Also, you have **D**BUILD there ;)

```
-DBUILD_STATIC=${BUILD_PLUGINS_STATIC}
-DBUILD_PLUGINS_STATIC=${BUILD_PLUGINS_STATIC}
```

File: ports/magnum/CONTROL [hunk 2]
```diff
@@ -1,42 +1,65 @@
...
+Feature: anyimageimporter
+Description: (Upcoming) AnyImageImporter plugin
+Build-Depends: magnum[trade]
 
-Feature: glutapplication
-Description: GlutApplication library
-Build-Depends: freeglut
+Feature: anyaudioimporter
+Description: (Upcoming) AnyAudioImporter plugin
+Build-Depends: magnum[trade]
```
Comment (mosra) [inline ports/magnum/CONTROL]: `magnum[audio]` instead

File: ports/magnum/CONTROL [hunk 3]
```diff
@@ -1,42 +1,65 @@
...
+Description: (Upcoming) gl-info utility
+Build-Depends: magnum[gl]
+
+Feature: glfwapplication
+Description: GlfwApplication library
+Build-Depends: glfw3
+
+Feature: glutapplication
+Description: GlutApplication library
+Build-Depends: freeglut
```
Comment (mosra) [inline ports/magnum/CONTROL]: `Build-Depends: freeglut, magnum[gl]`

File: ports/magnum/CONTROL [hunk 4]
```diff
@@ -46,24 +69,28 @@ Feature: magnumfontconverter
...
-Description: TgaImporter plugin
-
-Feature: debugtools
-Description: DebugTools library
-
-Feature: meshtools
-Description: MeshTools library
+Feature: opengltester
+Description: OpenGLTester library
+Build-Depends: corrade[testsuite]
```
Comment (mosra) [inline ports/magnum/CONTROL]: `Build-Depends: corrade[testsuite], magnum[gl]`

File: ports/magnum/CONTROL [hunk 5]
```diff
@@ -80,16 +107,19 @@ Build-Depends: magnum[texturetools]
...
+
+Feature: wavaudioimporter
+Description: WavAudioImporter plugin
 Build-Depends: magnum[audio]
 
 Feature: windowlesswglapplication
 Description: WindowlessWglApplication library
 
 Feature: wglcontext
 Description: WglContext library
```
Comment (mosra) [inline ports/magnum/CONTROL]: `Build-Depends: magnum[gl]`

File: ports/magnum/CONTROL [hunk 6]
```diff
@@ -80,16 +107,19 @@ Build-Depends: magnum[texturetools]
...
+
+Feature: trade
+Description: (Upcoming) Trade library
+
+Feature: wavaudioimporter
+Description: WavAudioImporter plugin
 Build-Depends: magnum[audio]
 
 Feature: windowlesswglapplication
 Description: WindowlessWglApplication library
```
Comment (mosra) [inline ports/magnum/CONTROL]: `Build-Depends: magnum[gl]`

File: ports/magnum/CONTROL [hunk 7]
```diff
@@ -1,42 +1,65 @@
...
+Feature: debugtools
+Description: DebugTools library
+
+Feature: distancefieldconverter
+Description: magnum-distancefieldconverter utility
+Build-Depends: magnum[texturetools]
 
 Feature: fontconverter
 Description: magnum-fontconverter utility
 Build-Depends: magnum[text]
```
Comment (mosra) [inline ports/magnum/CONTROL]: `Build-Depends: magnum[text], magnum[gl]`

File: ports/magnum/CONTROL [hunk 8]
```diff
@@ -1,42 +1,65 @@
...
 Build-Depends: openal-soft
 
-Feature: magnuminfo
-Description: magnum-info utility
+Feature: debugtools
+Description: DebugTools library
+
+Feature: distancefieldconverter
+Description: magnum-distancefieldconverter utility
+Build-Depends: magnum[texturetools]
```
Comment (mosra) [inline ports/magnum/CONTROL]: `Build-Depends: magnum[texturetools], magnum[gl]`

File: ports/magnum-plugins/CONTROL [hunk 9]
```diff
@@ -79,3 +87,6 @@ Feature: stbvorbisaudioimporter
 Description: StbVorbisAudioImporter plugin
 Build-Depends: magnum[audio]
 
+Feature: tinygltfimporter
+Description: (Upcoming) TinyGltfImporter plugin
```
Comment (mosra) [inline ports/magnum-plugins/CONTROL]: This is not upcoming anymore :)

File: ports/magnum-plugins/CONTROL [hunk 10]
```diff
@@ -1,31 +1,35 @@
 Source: magnum-plugins
-Version: 2018.02-1
-Build-Depends: magnum
+Version: 2018.02-2
```
Comment (mosra) [inline ports/magnum-plugins/CONTROL]: Why no `Build-Depends: magnum`?

File: scripts/buildsystems/msbuild/applocal.ps1 [hunk 11]
```diff
@@ -85,5 +93,11 @@ if (Test-Path "$g_install_root\bin\OpenNI2\openni2deploy.ps1") {
...
 
+# Note: This is a hack to make Magnum work.
+if (Test-Path "$g_install_root\bin\magnum\magnumdeploy.ps1") {
+    . "$g_install_root\bin\magnum\magnumdeploy.ps1"
+} elseif (Test-Path "$g_install_root\bin\magnum-d\magnumdeploy.ps1") {
+    . "$g_install_root\bin\magnum-d\magnumdeploy.ps1"
+}
+
 resolve($targetBinary)
-Write-Verbose $($g_searched | out-string)
```
Comment (mosra) [inline scripts/buildsystems/msbuild/applocal.ps1]: Why this was removed?

**Review by mosra — COMMENTED** [review]
Thanks a lot. :+1: 

Tricky question: does this automagically copy also dependencies of the plugins? (e.g. is libpng DLL that PngImporter depends on copied as well?) Or does vcpkg handle that part already? :)

**Comment by Squareys** [comment]
> does this automagically copy also dependencies of the plugins? (e.g. is libpng DLL that PngImporter depends on copied as well?) Or does vcpkg handle that part already? :)

Not so tricky answer: Yes! :D

File: ports/magnum-plugins/CONTROL (hunk 9 above)
Comment (Squareys) [reply ports/magnum-plugins/CONTROL]: Well, it's not in the release that has been tagged in the accompanying portfile.cmake, so "in perspective of the last release" it is indeed upcoming until that is tagged :)

File: ports/magnum-plugins/CONTROL (hunk 10 above)
Comment (Squareys) [reply ports/magnum-plugins/CONTROL]: I removed the magnum[trade] and didn't think :D

File: scripts/buildsystems/msbuild/applocal.ps1 (hunk 11 above)
Comment (Squareys) [reply scripts/buildsystems/msbuild/applocal.ps1]: Good question, thanks!

File: ports/magnum-plugins/CONTROL (hunk 9 above)
Comment (mosra) [reply ports/magnum-plugins/CONTROL]: Ah, sure, true, then leave it there. Sorry, my perception of time is mixed up again :)

**Comment by Squareys** [comment]
@mosra Done making the requested changes, thank you very much for the review!

@ras0219-msft This is now ready to merge, thanks for the help with https://github.com/mosra/magnum/issues/235 !

**Review by mosra — APPROVED** [review]

**Comment by Squareys** [comment]
@mosra Added the two missing feature dependencies, as discussed per gitter.

You may need to reapprove.

**Review by mosra — APPROVED** [review]

**Comment by Squareys** [comment]
@ras0219-msft This is ready to be merged. (In case it wasn't clear)

**Comment by ras0219-msft** [comment]
Thanks, this looks good to me as well!

**Pull request merged**